dev/DynamicDependency/M.AM.DD.PackageDependency.cpp

```diff
@@ -0,0 +1,105 @@
+
+#include <M.AM.Converters.h>
+
+namespace winrt::Microsoft::ApplicationModel::DynamicDependency::implementation
+{
+    PackageDependency::PackageDependency(PCWSTR id)
+    {
+        m_id = winrt::to_hstring(id);
+    }
+
+    PackageDependency::PackageDependency(hstring const& id) :
+        m_id(id)
+    {
+    }
+
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependency PackageDependency::GetFromId(hstring const& id)
+    {
+        return winrt::make<implementation::PackageDependency>(id);
+    }
+
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependency PackageDependency::Create(hstring const& packageFamilyName, Windows::ApplicationModel::PackageVersion const& minVersion)
+    {
+        const auto mddMinVersion = ::Microsoft::ApplicationModel::DynamicDependency::ToVersion(minVersion);
+        const auto mddArchitectures = MddPackageDependencyProcessorArchitectures::None;
+        const auto mddLifetimeKind = MddPackageDependencyLifetimeKind::Process;
+        const auto mddOptions = MddCreatePackageDependencyOptions::None;
+        return Create(packageFamilyName.c_str(), mddMinVersion, mddArchitectures, mddLifetimeKind, mddOptions);
+    }
+
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependency PackageDependency::Create(hstring const& packageFamilyName, Windows::ApplicationModel::PackageVersion const& minVersion, Microsoft::ApplicationModel::DynamicDependency::CreatePackageDependencyOptions const& options)
+    {
+        const auto mddMinVersion = ::Microsoft::ApplicationModel::DynamicDependency::ToVersion(minVersion);
+        const auto mddArchitectures = ::Microsoft::ApplicationModel::DynamicDependency::ToArchitectures(options.Architectures());
+        const auto mddLifetimeKind = ::Microsoft::ApplicationModel::DynamicDependency::ToLifetimeKind(options.LifetimeArtifactKind());
+        const auto mddOptions = ::Microsoft::ApplicationModel::DynamicDependency::ToCreateOptions(options);
+        return Create(packageFamilyName.c_str(), mddMinVersion, mddArchitectures, mddLifetimeKind, mddOptions);
+    }
+
+    Microsoft::ApplicationModel::DynamicDependency::PackageDependency PackageDependency::CreateForSystem(hstring const& packageFamilyName, Windows::ApplicationModel::PackageVersion const& minVersion, Microsoft::ApplicationModel::DynamicDependency::CreatePackageDependencyOptions const& options)
+    {
+        const auto mddMinVersion = ::Microsoft::ApplicationModel::DynamicDependency::ToVersion(minVersion);
+        const auto mddArchitectures = ::Microsoft::ApplicationModel::DynamicDependency::ToArchitectures(options.Architectures());
+        const auto mddLifetimeKind = ::Microsoft::ApplicationModel::DynamicDependency::ToLifetimeKind(options.LifetimeArtifactKind());
+        auto mddOptions = ::Microsoft::ApplicationModel::DynamicDependency::ToCreateOptions(options, MddCreatePackageDependencyOptions::ScopeIsSystem);
+        wil::unique_process_heap_string packageDependencyId;
```

Comment:
Doh! I really wish VC would warn about such things. Is there a way to tag a class to indicate only firing its ctor+dtor is OK no reason to warn, and anything w/o that marking should generate a warning if not otherwise accessed?

Reply:
I'm not sure. I'm not being as stylistically pedantic in this PR (trying to go easy on you :)) as I normally would be, but this is something you should avoid in general following [ES.21 Don't declare a variable until you have a value to initialize it with](http://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines.html#es22-dont-declare-a-variable-until-you-have-a-value-to-initialize-it-with). I would rather flag this type of thing were it unused or not.

Reply:
No worries. Feel free to share. I can always disagree :P

I'd say I agree about avoiding that in general, except I feel more strongly about that practice :-) It's C++ after all not C (and a feature going back decades). I've very fond of just-in-time local declaration

In this case it was a copy/paste leftover from earlier code. Maybe I'll take a look if one of the /Wall warnings would have flagged this and, if so, see how badly things go if I explicitly enable that 1 warning